processing-and-analysis/02-command-files/2s-practice-jackson/mlm-work-unique-ta-sects-jh.Rmd

```diff
@@ -58,140 +58,33 @@ master_unique <- na.omit(master_unique_2)
 
 ### Make `ta_sect` values unique across `course_fullid`
 
-#### First, I'm going to temporarily change `ta_sect` into character values
-
-```{r}
-master_unique$ta_sect <- as.character(master_unique$ta_sect)
+#### I'm going to be trying the code that CFC showed us
+
+```{r, eval = FALSE}
+master_unique_3 <- master_unique %>% 
+  select(course_fullid)
+master_unique_4 <- master_unique_3 %>% 
+  mutate(ta_sect = str_extract(course_fullid, "\\d\\d\\d\\d")) "\\d{4}"
```

Comment:
The `"\\d{4}"` at the end is one issue...that shouldn't be there. I wrote it last time just to show another way to write `"\\d\\d\\d\\d"`, but then forgot to delete it when I posted the code snippet in the agenda doc. (sorry)

But even if that weren't there, formally what this code would do is straight up replace the values in `ta_sect` with a string matching the pattern `"\\d\\d\\d\\d"` in `course_fullid`.